web/modules/custom/dpl_das/src/Plugin/rest/resource/DigitalArticleOrderResource.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Drupal\dpl_das\Plugin\rest\resource;
+
+use Drupal\Core\Config\ConfigManagerInterface;
+use Drupal\dpl_das\Elba\Client;
+use Drupal\dpl_das\Elba\Exception;
+use Drupal\dpl_das\Elba\PlaceCopyRequest;
+use Drupal\dpl_das\Input\ArticleOrder;
+use Drupal\dpl_login\LibraryAgencyIdProvider;
+use Drupal\rest\ModifiedResourceResponse;
+use Drupal\rest\Plugin\ResourceBase;
+use Drupal\rest\ResourceResponseInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\Serializer\Exception\UnexpectedValueException;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Resource for managing orders of digital articles.
+ *
+ * Digitizing articles is managed by the Royal Danish Library. Ordering an
+ * article is supported by the Elba webservices.
+ *
+ * Accessing this service requires a service account with an username and
+ * password and is thus not designed for direct browser based access by patron.
+ * Consequently it is exposed through the CMS instead.
+ *
+ * There is no canonical path for orders since this is not supported by the
+ * service. Patrons will have to consult their mail for confirmation that an
+ * order has been placed.
+ *
+ * @RestResource (
+ *   id = "dpl_das_digital_article_order",
+ *   label = @Translation("Digital Article Order"),
+ *   uri_paths = {
+ *     "create" = "/dpl_das/order",
+ *   },
+ *
+ *   payload = {
+ *     "name" = "order",
+ *     "description" = "Digital article order",
+ *     "in" = "body",
+ *     "schema" = {
+ *       "type" = "object",
+ *       "properties" = {
+ *         "pid" = {
+ *           "type" = "string",
+ *         },
+ *         "email" = {
+ *           "type" = "string",
+ *         },
+ *       },
+ *     },
+ *   },
+ *
+ *   responses = {
+ *     201 = {
+ *       "description" = "OK",
+ *     },
+ *     400 = {
+ *      "descriptions" = "Invalid input",
+ *     },
+ *     500 = {
+ *       "description" = "Internal server error",
+ *     },
+ *   }
+ * )
+ */
+class DigitalArticleOrderResource extends ResourceBase {
```

Comment:
I think we should have a test for the endpoint

Reply:
Good point. Based on the future of the integration do you want me to implement this?